<!DOCTYPE html>
    <html>
    <head>
        <meta charset="UTF-8">
        <title>PR2&colon; Curation of Radiolaria sequences</title>
        <style>
</style>
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.10.2/dist/katex.min.css" integrity="sha384-yFRtMMDnQtDRO8rLpMIKrtPCD5jdktao2TV19YiZYWMDkUR5GQZR/NOVTdquEx1j" crossorigin="anonymous">
<link href="https://cdn.jsdelivr.net/npm/katex-copytex@latest/dist/katex-copytex.min.css" rel="stylesheet" type="text/css">
        <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Microsoft/vscode/extensions/markdown-language-features/media/markdown.css">
<link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/Microsoft/vscode/extensions/markdown-language-features/media/highlight.css">
<style>
            body {
                font-family: -apple-system, BlinkMacSystemFont, 'Segoe WPC', 'Segoe UI', system-ui, 'Ubuntu', 'Droid Sans', sans-serif;
                font-size: 14px;
                line-height: 1.6;
            }
        </style>
        <style>
.task-list-item { list-style-type: none; } .task-list-item-checkbox { margin-left: -20px; vertical-align: middle; }
</style>
        
        <script src="https://cdn.jsdelivr.net/npm/katex-copytex@latest/dist/katex-copytex.min.js"></script>
        
    </head>
    <body class="vscode-body vscode-light">
        <h1 id="pr2-curation-of-radiolaria-sequences">PR2: Curation of Radiolaria sequences</h1>
<p>Miguel M. Sandin<br>
02-07-2020<br>
<a href="mailto:miguelmendezsandin@gmail.com">miguelmendezsandin@gmail.com</a></p>
<h2 id="dependencies">Dependencies</h2>
<ul>
<li><a href="https://mafft.cbrc.jp/alignment/software/">MAFFT</a> (Katoh and Standley, 2013).</li>
<li><a href="http://trimal.cgenomics.org/">trimAL</a> (Capella-Gutiérrez et al., 2009).</li>
<li><a href="https://cme.h-its.org/exelixis/web/software/raxml/index.html">RaxML</a> (Stamatakis, 2014).</li>
<li><a href="https://www.r-project.org/">R</a> (R Core Team, 2014).</li>
<li><a href="https://github.com/mothur/mothur/releases/tag/v.1.44.1">Mothur</a> (Schloss et al., 2009).</li>
<li><a href="https://github.com/torognes/vsearch">vsearch</a> (Rognes et al., 2016).</li>
<li><a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download">BLAST+</a></li>
</ul>
<h3 id="optional">Optional</h3>
<ul>
<li><a href="http://doua.prabi.fr/software/seaview">SeaView</a> (Gouy et al., 2010)</li>
<li><a href="http://www.ormbunkar.se/aliview/#DOWNLOAD">AliView</a> (Larsson, A. 2014)</li>
<li><a href="http://tree.bio.ed.ac.uk/software/figtree/">FigTree</a> (Rambaut 2016).</li>
<li><a href="https://rstudio.com/products/rstudio/download/">Rstudio</a>.</li>
</ul>
<hr>
<h2 id="step-1-building-a-detailed-reference-alignment">Step 1. Building a detailed reference alignment</h2>
<p>I have used as reference phylogenetic frameworks the last morpho-molecular classifications of <strong>Acantharea</strong> (Decelle et al. 2012), <strong>Collodaria</strong> (Biard et al. 2015), <strong>Nassellaria</strong> (Sandin et al. 2019) and <strong>Spumellaria</strong> (Sandin et al. 2020). Following these phylogenetic works I have selected 2 to 4-6 representative sequences of each clade (both morphologically described and environmental) to build a reference alignment (in total 211 sequences), in order to gather the most important and detailed genetic diversity. Such sequences were manually selected due to their high quality and their length, comprising the full 18S rDNA gene (when available). As showed in the reference phylogenetic frameworks, the partial 28S (D1+D2) rDNA gene was also used and concatenated in order to increase phylogenetic signal.</p>
<h3 id="11-align-sequences">1.1. Align sequences</h3>
<p>Sequences were aligned with MAFFT using the L-INS-i algorithm (‘--localpair’) and 1000 iterative refinement cycles for high accuracy:</p>
<pre><code><code><div>mafft --localpair --maxiterate 1000 FILE &gt; FILE_aligned
</div></code></code></pre>
<h3 id="12-check-alignment">1.2. Check alignment</h3>
<p>Alignment was manually check and corrected in SeaView (version 4.6.1) in order to align analogous positions and check for possible missalignments.</p>
<h3 id="13-trimm-alignment">1.3. Trimm alignment</h3>
<p>Corrected alignment matrix was trimmed with trimal selecting a 30% threshold:</p>
<pre><code><code><div>trimal -in FILE_alignedC -out FILE_alignedC_trimmed -gt 0.3
</div></code></code></pre>
<h3 id="14-phylogenetic-assessment">1.4. Phylogenetic assessment</h3>
<p>Two (semi-replicate) analysis were used to look for agreement -or not- and resolve possible dubious or low statistically supported groups/patterns using RAxML:</p>
<h4 id="141-gtrcat-as-substitution-model-and-1000-bootstraps-bs">1.4.1 GTR+CAT as substitution model and 1000 bootstraps (BS)</h4>
<pre><code><code><div>raxmlHPC-PTHREADS-SSE3 -T 8 -m GTRCAT -c 25 -p $RANDOM -x $(date +%s) -d -f a -N 1000 -n FILE_alignedC_trimmed_CAT -s FILE_alignedC_trimmed
</div></code></code></pre>
<h4 id="142-gtrgamma-with-1000-bs">1.4.2. GTR+GAMMA with 1000 BS</h4>
<pre><code><code><div>raxmlHPC-PTHREADS-SSE3 -T 8 -m GTRGAMMA -p $RANDOM -x $(date +%s) -d -f a -N $1000 -n FILE_alignedC_trimmed_GAMMA -s FILE_alignedC_trimmed
</div></code></code></pre>
<h3 id="15-re-ordering-alignment">1.5. Re-ordering alignment</h3>
<p>The resulting alignment was ordered following the phylogenetic tree in order to ease the manual correction of misalignments with the function ‘reorderFastaTree’ from the script ‘<a href="scripts/fastaFunctions.R">fastaFunctions.R</a>’ (<em>recommended to open it in R studio</em>).<br>
Steps <strong>1.2</strong> to <strong>1.5</strong> were repeated until the phylogenetic tree had a consistent topology and in agreement with previous studies (Decelle et al. 2012; Biard et al. 2015; Sandin et al. 2019; 2020). Phylogenetic trees were visualized in FigTree (version 1.4.3). Resulting alignment matrix was set as reference alignment for further steps.</p>
<p><sub><em>Note</em>: Steps <strong>1.1</strong> to <strong>1.3</strong> were carried out independently for 18S and 28S rDNA genes. Then both genes were concatenated for step <strong>1.4</strong> and further.</sub></p>
<hr>
<h2 id="step-2-creating-a-reference-backbone-phylogeny">Step 2. Creating a reference backbone phylogeny</h2>
<p>I continued adding the rest of publicly available and morphologically identified sequences detailed in the reference phylogenetic frameworks to the previous reference alignment (from <strong>step 1</strong>) using the function ‘--add’ from MAFFT and the FFT-NS-2 algorithm with a high gap opening penalty (‘--op 5’) as follows:</p>
<pre><code><code><div>mafft --thread 2 --inputorder --op 5.0 --add FILE2 --6merpair --maxiterate 1000 FILE_alignedC &gt; FILE2_aligned
</div></code></code></pre>
<p>And repeating steps <strong>1.2</strong> to <strong>1.5</strong> until consistent phylogenetic tree.</p>
<hr>
<h2 id="step-3-download-all-publicly-available-environmental-sequences">Step 3. Download all publicly available environmental sequences</h2>
<p>Reference 18S rDNA gene sequences from <strong>step 2</strong> were ‘blasted’ against NCBI to retrieve all publicly available environmental sequences of the 18S rDNA gene that belong to Radiolaria, as detailed in ‘Step 6’ of <a href="http://eukref.org/curation-pipeline-overview/">EukRef</a>. Previous versions of <a href="https://github.com/pr2database">PR2</a> (Guillou et al., 2013) were also blasted against NCBI in order to retrieve sequences not belonging to either Acantharea, Collodaria, Nassellaria or Spumellaria (i.e.; Rad-A, Rad-B, Rad-C or Radiolaria_X). Final dataset was checked in order to remove duplicates and other regions of the rDNA (except for the partial 28S rDNA gene that was left and concatenated, when available, to increase phylogenetic signal).</p>
<hr>
<h2 id="step-4-check-for-chimeras">Step 4. Check for chimeras</h2>
<p>Due to the difficulties in chimera detection, I have used two different algorithms implemented in:</p>
<h3 id="41-mothur">4.1. mothur</h3>
<pre><code><code><div>mothur &quot;#chimera.uchime(fasta=NEW, reference=REFERENCE)&quot;
</div></code></code></pre>
<h3 id="42-and-vsearch">4.2. and vsearch</h3>
<pre><code><code><div>vsearch --uchime_ref NEW --db REFERENCE --nonchimeras NEW_noChim --borderline NEW_poChim --chimeras NEW_Chim
</div></code></code></pre>
<p>Using as REFERENCE sequences the complete PR2 v4.11.0 database and as NEW those sequences extracted from <strong>step 3</strong>. Sequences detected as chimeras by any of the two methods were automatically removed and no longer considered in downstream analysis.</p>
<h3 id="43-blasting">4.3. Blasting</h3>
<p>A third in-house method was also used in order to detect as many chimeric or problematic sequences as possible. Here I blasted against NCBI independently the first and the last 300 bp of each sequence and compared the results. If there were less than 20 exact matches among the first 100 matches, the sequence was considered as <em>maybe</em> chimeric. If there are no exact matches among the first 100 matches, or it matches only with itself, the sequence was detected as <em>chimeric</em>. For further details see the script ‘<a href="scripts/chimera_blast.R">chimera_blast.R</a>’ (<em>recommended to open it in Rstudio</em>). Sequences detected either as maybe chimeric or chimeric were noted and manually checked in downstream phylogenetic assessments.</p>
<hr>
<h2 id="step-5-annotating-sequences">Step 5. Annotating sequences</h2>
<p>All environmental sequences resulting from <strong>step 4</strong> were automatically annotated with vsearch taking as reference sequences those morphologically described:</p>
<pre><code><code><div>vsearch --usearch_global ENVIRONMENTAL --db REFERENCE --blast6out OUTPUT --log OUTPUT_log
</div></code></code></pre>
<hr>
<h2 id="step-6-adding-environmental-sequences-to-the-backbone-phylogeny">Step 6. Adding environmental sequences to the backbone phylogeny</h2>
<p>Environmental annotated sequences from <strong>step 5</strong> were added to the reference alignment created in <strong>step 2</strong>. These sequences were gradually added according to their ‘reliability’ so it is easier to identify chimeric, <em>dubious</em>* or bad quality sequences:</p>
<ol>
<li>Firstly, were added those environmental sequences retrieved in previous studies and phylogenetically placed in a reference phylogenetic framework (Decelle et al. 2012; Biard et al. 2015; Sandin et al. 2019; 2020).</li>
<li>Secondly, were added these sequences that do not correspond within Acantharea, Collodaria, Nassellaria or Spumellaria (i.e.; Rad-A, Rad-B, Rad-C and Radiolaria_X).</li>
<li>Thirdly, and lastly, the rest of the sequences retrieved.</li>
</ol>
<p><sub>*I considered a dubious sequence a sequence that is clearly different (either in pair-wise similarity or phylogenetically) from the others and with a lack of similar sequences. In other words, if a sequence is different from others but has 2 or (ideally) more similar sequences, and preferably from different studies, is not considered as dubious.</sub></p>
<p>For every group of sequences that were added, the following steps were carried out:</p>
<h3 id="61-align-sequences-against-previously-created-reference-alignment">6.1. Align sequences against previously created reference alignment</h3>
<pre><code><code><div>mafft --thread 2 --inputorder --op 5.0 --add FILEX --6merpair --maxiterate 1000 FILE_alignedC &gt; FILEX_aligned
</div></code></code></pre>
<h3 id="62-trimming-alignment">6.2. Trimming alignment</h3>
<pre><code><code><div>trimal -in FILEX_aligned -out FILEX_aligned_trimmed -gt 0.3
</div></code></code></pre>
<h3 id="63-phylogenetic-assessment">6.3. Phylogenetic assessment</h3>
<p>Explore phylogenetic patterns, long branches, branches in unresolved positions, contrasting topology of previously resolved clades, …</p>
<h4 id="631-raxml-gtrcat-100-bs">6.3.1. RAxML GTR+CAT 100 BS</h4>
<pre><code><code><div>raxmlHPC-PTHREADS-SSE3 -T 16 -m GTRCAT -c 25 -p $RANDOM -x $(date +%s) -d -f a -N 100 -n FILEX_aligned_trimmed_CAT -s  FILEX_aligned_trimmed
</div></code></code></pre>
<h4 id="632-raxml-gtrgamma-100-bs">6.3.2. RAxML GTR+GAMMA 100 BS</h4>
<pre><code><code><div>raxmlHPC-PTHREADS-SSE3 -T 16 -m GTRGAMMA -p $RANDOM -x $(date +%s) -d -f a -N $100 -n FILEX_aligned_trimmed_GAMMA -s  FILEX_aligned_trimmed
</div></code></code></pre>
<h3 id="64-reorder-fasta">6.4. Reorder fasta</h3>
<p>Reordering fasta file with the function ‘reorderFastaTree’ from ‘fastaFunctions.R’, taking as reference tree that of 6.3.1 (or 6.3.2).</p>
<h3 id="65-manual-removing-of-problematic-sequences">6.5. Manual removing of problematic sequences</h3>
<p>Manual and final identification of chimeric, dubious or bad quality sequences for its deletion in AliView (version 1.26).
Repeat steps <strong>6.3</strong> to <strong>6.5</strong> until consistent phylogeny.</p>
<p><sub><em>Note</em>: in order to ease and speed <strong>step 6</strong>, after removing problematic sequences (<strong>6.5</strong>), I directly run the phylogenetic analysis (<strong>6.3</strong>), without realigning the raw sequences (<strong>6.1</strong>) and trimming the alignment (<strong>6.2</strong>). This is why I added a last confirmatory check:</sub></p>
<hr>
<h2 id="step-7-last-confirmatory-check">Step 7. Last confirmatory check</h2>
<p>All environmental sequences added in <strong>step 6</strong> that passed the stringent criteria were (re-)aligned to the backbone phylogenetic framework from <strong>step 2</strong> for a last phylogenetic assessment as described in steps <strong>6.1</strong> to <strong>6.3</strong>. And, if needed, step <strong>6.3</strong> was repeated after step <strong>6.4</strong> and manual checking and correction of the alignment in AliView (version 1.26).</p>
<hr>
<h2 id="step-8-final-taxonomic-annotation-and-corrections">Step 8. Final taxonomic annotation and corrections</h2>
<p>Final phylogenetic trees were used for the manual curation and correction of the taxonomic annotation of the enviromental sequences according to bootstrap support and phylogenetic relatedness to morphologically described sequences. Annotation was done according to the original description of:</p>
<ul>
<li><strong>Acantharea</strong> (Decelle et al. 2012): Clade A, Clade B, Clade C, Clade D, Clade E, Clade F and the environmental clades Acantharea 1, Acantharea 2 (subclades “a” and “b”), Acantharea 3 (subclades “a” and “b”) and Acantharea 4 (subclades “a” to “d”).<br>
<sub><em>Note</em>: Environmental clades of Acantharea were changed from roman numbers (I, II, III, IV) to Arabic numbers (1, 2, 3, 4 respectively) in order to homogenize the annotation.</sub></li>
<li><strong>Collodaria</strong> (Biard et al. 2015): Collophidiidae, Collosphaeridae and Sphaerozoidae, and some environmental sequences (annotated as Collodaria_X) as originally curated by Tristan Biard.</li>
<li><strong>Nassellaria</strong> (Sandin et al. 2019): Acanthodesmoidea (gathering Acanthodesmidae and Triospyrididae), Archipiliidae, Artostrobiidae, Carpocaniidae, Cycladophoridae, Eucyrtidiidae, Lychnocaniidae, Orosphaeridae (as described in Nakamura et al., 2020), Plagiacanthoidea (gathering Cannobotryidae, Lophophaenidae, Sethoperidae and Sethophormidae), Pterocorythidae and Theopiliidae and the environmental clades Nassellaria_1, Nassellaria_2, Nassellaria_3 and Nassellaria_4.<br>
<sub><em>Note1</em>: Acropyramioidea has no morphologically described representatives for the 18S rDNA gene, yet Nassellaria_1 is phylogenetically related to this clade.</sub><br>
<sub><em>Note2</em>: Nassellaria_3 shows very strong phylogenetic relationships with Artostrobiidae.</sub></li>
<li><strong>Spumellaria</strong> (Sandin et al. 2020): Actinommidae, Centrocuboidea (gathering Centrocubidae and Excentroconchidae), Hexastyloidea (gathering Hexalonchidae, Hexastylidae and Hollandosphaeridae), Spongodiscoidea (gathering Euchitoniidae and Spongobrachiidae), Spongopyloidea (gathering Cristallosphaeridae and Spongopylidae), Pylonioidea (gathering Pyloniidae and Tholoniidae), Astrosphaeridae, Lithelidae, Artiscinae, Rhizosphaeridae, Spongosphaeridae and Stylodictyidae and the envrionmental clades Spumellaria_1, Spumellaria_2, Spumellaria_3, Spumellaria_4 (subclades “a” and “b”) and Spumellaria_6.<br>
<sub><em>Note</em>: Spumellaria_5 (from Sandin et al. 2020) was emended within Liosphaeroidea (gathering Astrosphaeridae) as “Liosphaeroidea_X”.</sub></li>
</ul>
<p>Environmental clades (Rad-A, Rad-B, Rad-C and Radiolaria_X) were left as previously annotated, yet a more detailed taxonomy was applied when clear phylogenetic patterns (bootstrap ≥ 90 and consensus between GTR+GAMMA and GTR+CAT):</p>
<ul>
<li><strong>RAD-A</strong>: Clades from “a” to “g”.</li>
<li><strong>RAD-B</strong>: Group-I (subclades “a” to “f”), group-II, group-III and group-IV (with Taxopodida and subclades “a” to “e” and some nonassignable group of sequences “X”).</li>
<li><strong>RAD-C</strong>: Clades “a” and “b”.</li>
</ul>
<hr>
<h2 id="summary-and-concluding-remarks">Summary and concluding remarks</h2>
<p>In total, 4569 publicly available sequences associated to Radiolaria have been taxonomically curated and annotated <a href="files/pr2_Radiolaria_MMS.xlsx">(files/pr2_Radiolaria_MMS.xlsx)</a>. From these, 4556 are longer than 500bp and have less than 20 ambiguities <a href="files/pr2_rads_updated.fasta">(files/pr2_rads_updated.fasta)</a> and 289 sequences are new regarding older PR2 versions (as of v4.11.0). Main changes are:</p>
<ul>
<li><strong>Removing of chimeric, bad quality or dubious sequences</strong>: In total 391 sequences were manually removed, mainly from environmental clades (such as Acantharea_1, Acantharea_3, Rad-B, Radiolaria_X) and Liosphaeroidea (Spumellaria), among others.</li>
<li><strong>Updating Acantharea</strong> sequences after Decelle et al. (2012): 457 sequences.</li>
<li>Formal <strong>inclusion of Collodaria within Nassellaria</strong> (after Nakamura et al., 2020 and unpublished results from Miguel M. Sandin PhD thesis, 2019).</li>
<li><strong>Updating Nassellaria</strong> sequences after Sandin et al. (2019): 459 sequences, of which 190 formerly belonged to Collodaria.</li>
<li><strong>Updating Spumellaria</strong> sequences after Sandin et al. (2020): 3017 sequences.</li>
<li><strong>Finer taxonomic resolution of environmental clades</strong> RAD-A (107 sequences), RAD-B (455 sequences) and RAD-C (50 sequences).</li>
<li>Up to 24 sequences, phylogenetically related to Acantharia, were not possible to be assigned to any previously described clade, and were annotated as “<strong>Radiolaria_X</strong>”.</li>
</ul>
<p>Here we built a backbone phylogenetic framework of all the described diversity of Radiolaria to which we have aligned the rest of publicly available environmental sequences, after removing problematic sequences (chimeras, dubious or bad quality). All this gathers fully reliable sequences representing the biggest part of the diversity of Radiolaria, that we can trust and use for phylogenetic reconstruction, phylogenetic placement of short reads or annotation of the still to come full length environmental rDNA sequences from Oxford Nanopore Technologies or Pac-Bio.</p>
<p>Lastly, I would like to mention that taxonomic hierarchy and levels may differ from one reference to another. For example in Cavalier-smith (2017) Polycystinea (gathering Spumellaria, Nassellaria and Collodaria) holds a “superclass” position along with Spasmaria (gathering Acantharea and Sticholonche). On the other side, in Adl et al. (2019) Polycystinea holds a “class” position along with Acantharea and Taxopodida. You may therefore select specific taxa according to the scope of your study bearing in mind these differences in taxonomic hierarchy and levels.</p>
<hr>
<h2 id="references">References</h2>
<ul>
<li>Adl, S. M., Bass, D., Lane, C. E., Lukeš, J., Schoch, C. L., Smirnov, A., et al. 2019. Revisions to the classiﬁcation, nomenclature, and diversity of eukaryotes. J. Eukaryot. Microbiol. 66, 4–119. <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/jeu.12691">doi:10.1111/jeu.12691</a></li>
<li>Biard, T., Bigeard, E., Audic, S., Poulain, J., Stemmann, L., Not, F., 2017. Biogeography and diversity of Collodaria (Radiolaria) in the global ocean. Nat. Publ. Gr. 1–42. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5437347/pdf/ismej201712a.pdf">doi:10.1038/ismej.2017.12</a></li>
<li>Cavalier-Smith, T., Chao, E.E., Lewis, R., 2018. Multigene phylogeny and cell evolution of chromist infrakingdom Rhizaria: contrasting cell organisation of sister phyla Cercozoa and Retaria. Protoplasma 255, 1517–1574. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6133090/">doi:10.1007/s00709-018-1241-1</a></li>
<li>Capella-Gutiérrez, S., Silla-Martínez, J.M., Gabaldón, T., 2009. trimAl: A tool for automated alignment trimming in large-scale phylogenetic analyses. Bioinformatics 25, 1972–1973. <a href="https://academic.oup.com/bioinformatics/article/25/15/1972/213148">doi:10.1093/bioinformatics/btp348</a></li>
<li>Decelle, J., Suzuki, N., Mahé, F., Vargas, C. De, Not, F., 2012b. Molecular Phylogeny and Morphological Evolution of the Acantharea (Radiolaria). Protist 163, 435–450. <a href="https://www.sciencedirect.com/science/article/abs/pii/S1434461011000988?via%3Dihub">doi:10.1016/j.protis.2011.10.002</a></li>
<li>Gouy, M., Guindon, S., Gascuel, O., 2010. SeaView Version 4: A Multiplatform Graphical User Interface for Sequence Alignment and Phylogenetic Tree Building. Mol. Biol. Evol. 27, 221–224. <a href="https://academic.oup.com/mbe/article/27/2/221/970247">doi:10.1093/molbev/msp259</a></li>
<li>Guillou, L., Bachar, D., Audic, S., Bass, D., Berney, C., Bittner, L., Boutte, C., Burgaud, G., De Vargas, C., Decelle, J., Del Campo, J., Dolan, J.R., Dunthorn, M., Edvardsen, B., Holzmann, M., Kooistra, W.H.C.F., Lara, E., Le Bescot, N., Logares, R., Mahé, F., Massana, R., Montresor, M., Morard, R., Not, F., Pawlowski, J., Probert, I., Sauvadet, A.L., Siano, R., Stoeck, T., Vaulot, D., Zimmermann, P., Christen, R., 2013. The Protist Ribosomal Reference database (PR2): A catalog of unicellular eukaryote Small Sub-Unit rRNA sequences with curated taxonomy. Nucleic Acids Res. 41, 597–604.<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3531120/pdf/gks1160.pdf">doi:10.1093/nar/gks1160</a></li>
<li>Katoh, K., Standley, D.M., 2013. MAFFT multiple sequence alignment software version 7: Improvements in performance and usability. Mol. Biol. Evol. 30, 772–780. <a href="https://academic.oup.com/mbe/article/30/4/772/1073398">doi:10.1093/molbev/mst010</a></li>
<li>Larsson, A. (2014). AliView: a fast and lightweight alignment viewer and editor for large data sets. Bioinformatics30(22): 3276-3278. <a href="https://academic.oup.com/bioinformatics/article/30/22/3276/2391211">http://dx.doi.org/10.1093/bioinformatics/btu531</a></li>
<li>Nakamura, Y., Sandin, M.M., Suzuki N., Somiya R., Tuji A., Not, F, 2020. Phylogenetic revision of the order Entactinaria - Paleozoic relict Radiolaria (Rhizaria, SAR). Protist 171. <a href="https://reader.elsevier.com/reader/sd/pii/S1434461019300707?token=5DD9A1B3581FD91E73FDF99BF683DCF68658895F7A186C03566C0C1146EA7F8FB4FDB2E76FFA7D188CA812DFEFB57B9D">doi:10.1016/j.protis.2019.125712</a></li>
<li>R Core Team (2019). R: A language and environment for statistical computing. R Foundation for Statistical Computing, Vienna, Austria. URL <a href="https://www.r-project.org/">https://www.R-project.org/</a></li>
<li>Rambaut A (2016) FigTree version 1.4.3. <a href="http://tree.bio.ed.ac.uk/software/figtree/">http://tree.bio.ed.ac.uk/software/figtree/</a></li>
<li>Rognes, T., Flouri, T., Nichols, B., Quince, C., Mahé, F., 2016. VSEARCH: a versatile open source tool for metagenomics. PeerJ 4, e2584.<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5075697/pdf/peerj-04-2584.pdf">doi:10.7717/peerj.2584</a></li>
<li>Sandin M.M., Biard T., Romac S., O’Dogherty L., Suzuki N., Not F., 2020. Morpho-molecular diversity and evolutionary analyses suggest hidden life styles in Spumellaria (Radiolaria) bioRxiv 2020.06.29.176917; doi: <a href="https://www.biorxiv.org/content/10.1101/2020.06.29.176917v1">https://doi.org/10.1101/2020.06.29.176917</a></li>
<li>Sandin, M.M., Pillet, L., Biard, T., Poirier, C., Bigeard, E., Romac, S., Suzuki, N., Not, F., 2019. Time Calibrated Morpho-molecular Classification of Nassellaria (Radiolaria). Protist 170, 187–208. <a href="https://reader.elsevier.com/reader/sd/pii/S143446101830110X?token=5A45B3EAB7E3260253DBBDAC41398181BF8FEB3E40B4CD9F7BEF12B3DF4312F96D955633C3DB076EA70E8D77242FE6F3">doi:10.1016/j.protis.2019.02.002</a></li>
<li>Schloss, P.D., Westcott, S.L., Ryabin, T., Hall, J.R., Hartmann, M., Hollister, E.B., Lesniewski, R.A., Oakley, B.B., Parks, D.H., Robinson, C.J., Sahl, J.W., Stres, B., Thallinger, G.G., Van Horn, D.J., Weber, C.F., 2009. Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl. Environ. Microbiol. 75, 7537–7541.<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2786419/pdf/1541-09.pdf">doi:10.1128/AEM.01541-09</a></li>
<li>Stamatakis, A., 2014. RAxML version 8: A tool for phylogenetic analysis and post-analysis of large phylogenies. Bioinformatics 30, 1312–1313. <a href="https://academic.oup.com/bioinformatics/article/30/9/1312/238053">doi:10.1093/bioinformatics/btu033</a></li>
</ul>

    </body>
    </html>